pkg/cache/context.go

```diff
@@ -765,3 +767,62 @@ func (ctx *Context) SchedulerNodeEventHandler() func(obj interface{}) {
 	// this is not required in some tests
 	return nil
 }
+
+func createConfigMapSelector() (labels.Selector, error) {
+	slt := labels.NewSelector()
+	appVals := [] string {"yunikorn"}
+	req, err := labels.NewRequirement(common.LabelApp, selection.Equals, appVals)
+	if err != nil {
+		return  nil, err
+	}
+	slt = slt.Add(*req)
+	return slt, nil
+}
+
+func findYKConfigMap(configMaps []*v1.ConfigMap) (*v1.ConfigMap, error) {
+	var ykconf *v1.ConfigMap
+	foundConf := false
+	if len(configMaps) == 0 {
+		return nil, fmt.Errorf("configmap with label app:yunikorn not found")
+	}
+	for _, c := range configMaps {
+		if c.Name == common.DefaultConfigMapName {
+			ykconf = c
+			foundConf = true
+			break
+		}
+	}
+	if !foundConf {
+		return nil, fmt.Errorf("configmap with name %s not found", common.DefaultConfigMapName)
+	}
+	return ykconf, nil
+}
```

Comment:
this can be simplified to:

```
for _, c := range configMaps {
  if c.Name == common.DefaultConfigMapName {
    return c
  }
}
return nil, fmt.Errorf("configmap with name %s not found", common.DefaultConfigMapName)
```
because range handles the case when the slice is nil